packages/eas-cli/src/commands/workflow/runs.ts

```diff
@@ -0,0 +1,151 @@
+import { Flags } from '@oclif/core';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EasJsonOnlyFlag } from '../../commandUtils/flags';
+import { getLimitFlagWithCustomValues } from '../../commandUtils/pagination';
+import { WorkflowRunFragment, WorkflowRunStatus } from '../../graphql/generated';
+import { AppQuery } from '../../graphql/queries/AppQuery';
+import { WorkflowRunQuery } from '../../graphql/queries/WorkflowRunQuery';
+import Log from '../../log';
+import formatFields from '../../utils/formatFields';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export type WorkflowRunResult = {
+  id: string | null;
+  status: string | null;
+  gitCommitMessage?: string | null;
+  gitCommitHash?: string | null;
+  startedAt: string | null;
+  finishedAt: string | null;
+  workflowId: string | null;
+  workflowName: string | null;
+  workflowFileName: string | null;
+};
+
+function processWorkflowRuns(
+  runs: WorkflowRunFragment[],
+  params: {
+    workflowFileName?: string | undefined;
+    status?: string | undefined;
+  }
+): WorkflowRunResult[] {
+  const { workflowFileName, status } = params;
+  return runs
+    .filter(run => {
+      if (workflowFileName && run.workflow?.fileName !== workflowFileName) {
+        return false;
+      }
+      if (status && run.status !== status) {
+        return false;
+      }
+      return true;
+    })
+    .map(run => {
+      const finishedAt = run.status === WorkflowRunStatus.InProgress ? null : run.updatedAt;
+      return {
+        id: run.id ?? null,
+        status: run.status ?? null,
+        gitCommitMessage: run.gitCommitMessage ?? null,
+        gitCommitHash: run.gitCommitHash ?? null,
+        startedAt: run.createdAt ?? null,
+        finishedAt,
+        workflowId: run.workflow?.id ?? null,
+        workflowName: run.workflow?.name ?? null,
+        workflowFileName: run.workflow?.fileName ?? null,
+      };
+    });
+}
```

Comment:
no need for client side filtering any more, also can improve nullability of types that are never null now that the common fragment input is used

```suggestion
type WorkflowRunResult = {
  id: string;
  status: string;
  gitCommitMessage: string | null;
  gitCommitHash: string | null;
  startedAt: string;
  finishedAt: string;
  workflowId: string;
  workflowName: string | null;
  workflowFileName: string;
};

function processWorkflowRuns(runs: WorkflowRunFragment[]): WorkflowRunResult[] {
  return runs.map(run => {
    const finishedAt = run.status === WorkflowRunStatus.InProgress ? null : run.updatedAt;
    return {
      id: run.id,
      status: run.status,
      gitCommitMessage: run.gitCommitMessage ?? null,
      gitCommitHash: run.gitCommitHash ?? null,
      startedAt: run.createdAt,
      finishedAt,
      workflowId: run.workflow.id,
      workflowName: run.workflow.name ?? null,
      workflowFileName: run.workflow.fileName,
    };
  });
}
```